dap-mode.el

```diff
@@ -1668,10 +1674,27 @@ after selecting configuration template."
                             (funcall debug-args))
                           (user-error "Have you loaded the `%s' specific dap package?"
                                       (or (plist-get debug-args :type)
-                                          (user-error "%s does not specify :type" debug-args))))))
-    (if (functionp launch-args)
-        (funcall launch-args #'dap-start-debugging-noexpand)
-      (dap-start-debugging-noexpand launch-args))))
+                                          (user-error "%s does not specify :type" debug-args)))))
+         (cb (lambda ()
+               (if (functionp launch-args)
+                   (funcall launch-args #'dap-start-debugging-noexpand)
+                 (dap-start-debugging-noexpand launch-args)))))
+    (-if-let ((&plist :compilation) launch-args)
+        (progn
+          (cl-remf launch-args :compilation)
```

Comment:
nit: can you move it in `dap-start-debugging-noexpand` near the other cleanup? Also, I think that it will be better if we can use :dap-compilation to avoid collisions with valid keys template keys.

Reply:
Are you sure it `dap-start-debugging-noexpand` is the right function for this? Isn't it supposed to be a more low-level interface?

Reply:
Second point is fixed.

Reply:
it already deletes the other custom stuff like  `:cleanup-function` it seems like it is better to keep the cleanup in one place.

Reply:
The problem with using :cleanup-function is that it would have to have a closure over the compilation window (since we don't want to delete the compilation buffer if the user opens it manually), and that wouldn't work with `dap-debug-last`.

Reply:
I am talking solely about removing :compilation key from the plist.

Reply:
... and the location of that code. 

> (cl-remf launch-args :compilation)

Reply:
Ahh, I understand now. However, `:dap-compilation` wouldn't be treated specially unlike the other fields when using `dap-start-debugging-noexpand`, so I think it makes sense to only remove it in `dap-debug`.

Reply:
My understanding is  that we remove keys to avoid sending them to the debug adapter. ATM this is happening in `dap-start-debugging-noexpand`. Also, you should not remove it inplace from the configuration othewise it will be removed from the rerun.

Reply:
We wouldn't modify the configuration, because `dap-variables-expand-in-launch-configuration` necessarily copies the tree passed to it. I just realized that we are needlessly copying the conf in the `interactive` block.